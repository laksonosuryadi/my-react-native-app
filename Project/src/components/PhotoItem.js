import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  TouchableHighlight,
  Dimensions,
} from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const PhotoItem = (props) => {
  const { navigate } = props.navigation
  return(
    <View style={styles.content}>
      <TouchableHighlight onPress={() => navigate('Detail', {detail: props.photo})}>
      <Image
          style={{width: deviceWidth * 0.327, height: '100%'}}
          source={{uri: props.photo.image_url}}
        />
      </TouchableHighlight>
    </View>
  )
}

const styles = {
  content: {
    marginLeft: 1,
    marginRight: 1,
    marginLeft: 1,
    marginBottom: 0,
    justifyContent: 'center',
    alignItems: 'center',
    width: deviceWidth * 0.327,
    height: deviceWidth * 0.327,
    borderWidth: 1,
    borderColor: 'white',
    position: 'relative'
  }
}

export default PhotoItem
