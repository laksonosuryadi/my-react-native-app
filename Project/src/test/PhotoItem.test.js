import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';

import PhotoItem from '../components/PhotoItem'

it('render PhotoItem successfully', () => {
  const tree = renderer.create(
      <PhotoItem
        navigation={{
          navigate: function() {}
        }}
        photo={{
          image_url: ''
        }}
      />
  )
  expect(tree).toMatchSnapshot();
})
