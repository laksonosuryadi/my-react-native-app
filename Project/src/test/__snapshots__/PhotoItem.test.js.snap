// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PhotoItem successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "white",
      "borderWidth": 1,
      "height": 245.25,
      "justifyContent": "center",
      "marginBottom": 0,
      "marginLeft": 1,
      "marginRight": 1,
      "position": "relative",
      "width": 245.25,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": 245.25,
        }
      }
    />
  </View>
</View>
`;
